plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +196,465 @@ in
         ```
       '';
 
-      showCloseIcon = helpers.defaultNullOpts.mkBool true "Whether to show the close icon.";
+      show_close_icon = defaultNullOpts.mkBool true ''
+        Whether to show the close icon.
+      '';
 
-      showTabIndicators = helpers.defaultNullOpts.mkBool true "Whether to show the tab indicators.";
+      show_tab_indicators = defaultNullOpts.mkBool true ''
+        Whether to show the tab indicators.
+      '';
 
-      showDuplicatePrefix = helpers.defaultNullOpts.mkBool true "Whether to show the prefix of duplicated files.";
+      show_duplicate_prefix = defaultNullOpts.mkBool true ''
+        Whether to show the prefix of duplicated files.
+      '';
 
-      enforceRegularTabs = helpers.defaultNullOpts.mkBool false "Whether to enforce regular tabs.";
+      duplicates_across_groups = defaultNullOpts.mkBool true ''
+        Whether to consider duplicate paths in different groups as duplicates.
+      '';
 
-      alwaysShowBufferline = helpers.defaultNullOpts.mkBool true "Whether to always show the bufferline.";
+      enforce_regular_tabs = defaultNullOpts.mkBool false ''
+        Whether to enforce regular tabs.
+      '';
 
-      persistBufferSort = helpers.defaultNullOpts.mkBool true "Whether to make the buffer sort persistent.";
+      always_show_bufferline = defaultNullOpts.mkBool true ''
+        Whether to always show the bufferline.
+      '';
 
-      maxPrefixLength = helpers.defaultNullOpts.mkInt 15 "Maximum prefix length";
+      auto_toggle_bufferline = defaultNullOpts.mkBool true ''
+        Whether to automatically toggle bufferline.
+      '';
 
-      sortBy = helpers.defaultNullOpts.mkStr "id" "sort by";
+      persist_buffer_sort = defaultNullOpts.mkBool true ''
+        Whether to make the buffer sort persistent.
+      '';
 
-      diagnostics = helpers.defaultNullOpts.mkNullable (
-        with types;
-        either bool (enum [
-          "nvim_lsp"
-          "coc"
-        ])
-      ) false "diagnostics";
+      move_wraps_at_ends = defaultNullOpts.mkBool true ''
+        Whether or not the move command "wraps" at the first or last position.
+      '';
 
-      diagnosticsIndicator = helpers.defaultNullOpts.mkLuaFn null "Either `null` or a function that returns the diagnostics indicator.";
+      max_prefix_length = defaultNullOpts.mkInt 15 ''
+        Maximum prefix length used when a buffer is de-duplicated.
+      '';
 
-      diagnosticsUpdateInInsert = helpers.defaultNullOpts.mkBool true "Whether diagnostics should update in insert mode";
+      sort_by =
+        defaultNullOpts.mkNullableWithRaw
+          (
+            with types;
+            oneOf [
+              (enum [
+                "insert_after_current"
+                "insert_at_end"
+                "id"
+                "extension"
+                "relative_directory"
+                "directory"
+                "tabs"
+              ])
+              rawLua
+            ]
+          )
```

Comment:
I'm confused, is this a typo?

We have `mkNullableWithRaw`, which wraps the type in `maybeRaw` (aka `either rawLua`) but then we also have a `oneOf [ rawLua ]`?

Reply:
Looks like it got left in there when I did mkNullable first. I'll change it over

Reply:
Was it related to this https://github.com/nix-community/nixvim/pull/2013#discussion_r1715172032?

Reply:
That probably could have thrown me, too.